dashboard/src/actions/profileActions.js

```
@@ -0,0 +1,92 @@
+
+    if (response.status === 200 && Object.keys(response.data).length > 0) {
+      dispatch({
+        type: TYPES.GET_USER_DETAILS,
+        payload: response?.data,
+      });
+    } else {
+      let toast = {
+        variant: "danger",
+        title: response?.data?.message,
+      };
+      dispatch({
+        type: TYPES.SHOW_TOAST,
+        payload: { ...toast },
+      });
+    }
+    dispatch({ type: TYPES.COMPLETED });
+  } catch (error) {
+    dispatch({ type: TYPES.NETWORK_ERROR });
+    dispatch({ type: TYPES.COMPLETED });
+  }
+};
+
+export const updateUserDetails =
+  (value, fieldName) => async (dispatch, getState) => {
+    let userDetails = { ...getState().userProfile.userDetails };
+    userDetails[fieldName] = value;
+
+    dispatch({
+      type: TYPES.UPDATE_USER_DETAILS,
+      payload: { ...userDetails },
+    });
+  };
+
+export const sendForUpdation = () => async (dispatch, getState) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+    const userDetails = { ...getState().userProfile.userDetails };
+    const userDetails_copy = { ...getState().userProfile.userDetails_copy };
+
+    let keys = ["first_name", "last_name", "email"];
+    let params = {};
+    for (let key of keys) {
+      if (userDetails[key] !== userDetails_copy[key]) {
+        params[key] = userDetails[key];
+      }
+    }
+    const username = getState().userAuth.loginDetails.username;
+    if (username) {
+      const response = await API.put(`${API_ROUTES.GET_USER}/${username}`, {
+        ...params,
+      });
+      if (response.status === 200) {
+        let toast = {
+          variant: "success",
+          title: "Updated!",
+        };
+        dispatch({
+          type: TYPES.SHOW_TOAST,
+          payload: { ...toast },
+        });
+      }
+      dispatch({ type: TYPES.COMPLETED });
+      return true;
+    }
+  } catch (error) {
+    dispatch({ type: TYPES.COMPLETED });
+  }
```

So on user update we're just ignoring errors completely?